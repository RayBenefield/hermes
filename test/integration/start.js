import flame from '@leonardvandriel/flame';
import * as databases from './database'; // eslint-disable-line import/no-unresolved,import/extensions,no-unused-vars
import uuid from '../utils/mock-uuid';
import random from '../utils/mock-random';
import configureSave from '../utils/save-db';
import configureDomain from '../../src/domain';
import configureFlame from '../../src/lib/local-db';

const db = configureFlame(flame);
const domain = configureDomain({ db, uuid, random });
const save = configureSave(flame); // eslint-disable-line no-unused-vars
beforeEach(() => flame.loadDatabase(undefined));

test('notify all players of new game', () => {
    // Given
    flame.loadDatabase(databases['new-game']);

    // When
    return domain({
        action: 'start',
        game: {
            id: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
            players: {
                123456: true,
                234567: true,
                345678: true,
                456789: true,
                567890: true,
            },
            notified_players: {
                567890: true,
            },
        },
    })
        .then(save('new-round'))
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});
