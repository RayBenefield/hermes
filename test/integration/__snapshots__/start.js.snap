// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notify all players of new game 1`] = `
Object {
  "db": Object {
    "games": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        "notified_players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
      },
    },
    "hands": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "123456": Object {
          "cards": Object {
            "278": true,
            "279": true,
            "280": true,
            "281": true,
            "282": true,
            "284": true,
            "285": true,
            "286": true,
            "287": true,
            "415": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 123456,
        },
        "234567": Object {
          "cards": Object {
            "19": true,
            "20": true,
            "22": true,
            "23": true,
            "415": true,
            "543": true,
            "544": true,
            "545": true,
            "546": true,
            "547": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 234567,
        },
        "345678": Object {
          "cards": Object {
            "151": true,
            "152": true,
            "153": true,
            "413": true,
            "414": true,
            "415": true,
            "694": true,
            "695": true,
            "915": true,
            "917": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 345678,
        },
        "456789": Object {
          "cards": Object {
            "217": true,
            "348": true,
            "415": true,
            "479": true,
            "760": true,
            "850": true,
            "851": true,
            "86": true,
            "981": true,
            "982": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 456789,
        },
        "567890": Object {
          "cards": Object {
            "1014": true,
            "119": true,
            "381": true,
            "415": true,
            "512": true,
            "596": true,
            "662": true,
            "727": true,
            "818": true,
            "949": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 567890,
        },
      },
    },
    "players": Object {
      "facebook": Object {
        "123456": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 123456,
          "platform": "facebook",
        },
        "234567": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 234567,
          "platform": "facebook",
        },
        "345678": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 345678,
          "platform": "facebook",
        },
        "456789": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 456789,
          "platform": "facebook",
        },
        "567890": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 567890,
          "platform": "facebook",
        },
      },
    },
    "queue": Object {},
    "rounds": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        },
      },
    },
  },
  "results": Object {
    "action": "start",
    "game": Object {
      "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
      "notified_players": Object {
        "123456": true,
      },
      "players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
    },
    "messages": Array [
      Object {
        "payload": Object {
          "players": Array [
            Object {
              "first_name": "Ray",
              "id": "123456",
            },
            Object {
              "first_name": "Ray",
              "id": "234567",
            },
            Object {
              "first_name": "Ray",
              "id": "345678",
            },
            Object {
              "first_name": "Ray",
              "id": "456789",
            },
            Object {
              "first_name": "Ray",
              "id": "567890",
            },
          ],
        },
        "type": "notify-game-started-message",
      },
    ],
    "players": Array [
      Object {
        "first_name": "Ray",
        "id": 123456,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "id": 234567,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "id": 345678,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "id": 456789,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "id": 567890,
        "platform": "facebook",
      },
    ],
    "unnotifiedPlayers": Array [
      Object {
        "first_name": "Ray",
        "id": 234567,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "id": 345678,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "id": 456789,
        "platform": "facebook",
      },
      Object {
        "first_name": "Ray",
        "id": 567890,
        "platform": "facebook",
      },
    ],
  },
}
`;
